src/ostorlab/agent/agent.py

```diff
@@ -244,6 +243,15 @@ def process_cleanup(self) -> None:
         """
         raise NotImplementedError()
 
+    def _at_exit(self, *args) -> None:
```

Comment:
I think the private method  _at_exit should be renamed since it's confused with the public one _at_exit.
why not _terminate_and_cleanup